nixos/incus: support bzip2, lz4 and zstd compression formats

djds opened the pull request:
replace environment.PATH with path

djds committed:
nixos/incus: add zstd to serverBinPath

zstd is a supported compression algo for images:
https://linuxcontainers.org/incus/docs/main/server_config/#server-options-images

djds committed:
nixos/incus: replace environment.PATH with path

Allows users to amend the path without updating the module.

djds commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 466666`
Commit: `b2476c6feb2c1266a64a583b5e36c1ed3c7c8435`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>

adamcstephens commented:
You can't run nixpkgs-review on a module change. In this case you should build incus-lts.tests and incus.tests

djds commented:
> You can't run nixpkgs-review on a module change. In this case you should build incus-lts.tests and incus.tests

Yeah lol I realized that as soon as I did: force of habit.

adamcstephens reviewed (changes requested):
We shouldn't need this everywhere, please adjust. You'll also have to revert the removal of the prepended path string due to qemu_kvm not having a bin directory.

djds commented on nixos/modules/virtualisation/incus.nix:
```diff
@@ -441,7 +446,10 @@ in
-      inherit environment;
+      inherit
+        environment
+        path
```
The original `inherit environment;` call would include `environment.PATH`, so this is minimal change unless I'm misunderstanding.

djds commented on nixos/modules/virtualisation/incus.nix:
```diff
@@ -11,78 +11,78 @@ let
+    ++ lib.optionals config.virtualisation.vswitch.enable [ config.virtualisation.vswitch.package ]
+    ++ lib.optionals config.boot.zfs.enabled [
+      config.boot.zfs.package
+      "${config.boot.zfs.package}/lib/udev"
```
Doesn't that mean this would never have worked per https://github.com/NixOS/nixpkgs/pull/466666/files#r2577404366?

adamcstephens replied:
Hmm, maybe not. But I've never noticed and am a zfs user.

djds replied:
Fixed this with the `linkFarm` call.

adamcstephens commented on nixos/modules/virtualisation/incus.nix:
```diff
@@ -420,7 +422,10 @@ in
-      inherit environment;
+      inherit
+        environment
+        path
```
Fair enough, but that was probably a mistake originally. If we remove it from these two services do the tests pass? If so, then we should not carry it along. Including this path brings a lot of dependencies into scope for these services, and increases the churn their (admittedly small) unit derivations will have. Either way, it should be simple to test that we don't need them.

adamcstephens committed:
nixos/incus: support bzip2 and lz4 compression

These are supported upstream, so provide full support here:

https://github.com/lxc/incus/blob/a229a7cd59e8be258991808e4dce68b130c128ab/shared/archive/detect.go#L39

adamcstephens reviewed (approved):
built tests for incus and incus-lts on aarch64-linux
